### PR TITLE
Add a test that tries to use the generated CMake libs [continuation].

### PR DESCRIPTION
**Pull request by dmitrykos**
It is a continuation of stalled (and thus closed) PR #671.

### CODE REVIEW

**Comment by dmitrykos** [comment]
Besides fix of #671 this PR also fixes CI pipeline due to outdated vcpkg commit Id that was causing MSVC and MinGW build failure although was working well before. The reason was missing dependency `msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst` of vcpkg.

Now it is set to the latest (https://github.com/microsoft/vcpkg/commit/ca9ac0ba65965937fb66783c4f726c2c755ad9d9) and works well.

**Comment by RossBencina** [comment]
Does this remove the vcpkg dependency on Mac CI? (i.e. does it close #777)

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -10,28 +10,23 @@ jobs:
...
             asio_sdk_cache_path: asiosdk.zip
             dependencies_extras: mingw-w64
-            # FIXME: linking JACK fails with vcpkg. Switch the CMake toolchain file to use vcpkg. The
-            # toolchain file in this repository is not needed when using a MinGW triplet with vcpkg.
             vcpkg_triplet: x64-mingw-static
             cmake_generator: "Unix Makefiles"
+            # FIXME: linking vcpkg Jack fais due to MinGW-W64 linker assertion. Disable Jack for this build.
             cmake_options:
               -DPA_USE_ASIO=ON
+              -DPA_USE_JACK=OFF
```
Comment (RossBencina) [inline .github/workflows/cmake.yml:26]: Could you create a ticket regarding this issue please?

**Review by philburk — APPROVED** [review]
Thanks for figuring this out!
I don't fully understand it but CI is passing.
So that is a good thing.

**Review by RossBencina — APPROVED** [review]
Thanks Dmitry. I don't understand what you've done here but I'm OK moving forward with this.

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (dmitrykos) [reply .github/workflows/cmake.yml:26]: There is linker error, so it is outside PA and either Jack dependency must be fixed or MinGW-64 linker has to be changed. I can create ticket to trace this issue although we could simply revise `DPA_USE_JACK=OFF` after some time.

**Pull request merged**